cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -72,6 +77,32 @@ using v1::by_default;
 
 } // namespace method
 
+/// Represents result option flag
+/// Behaves like a regular :expr`enum`.
+class result_option_id : public result_option_id_base {
+public:
+    constexpr result_option_id() : result_option_id_base{} {}
+    constexpr result_option_id(const result_option_id_base& base) : result_option_id_base{ base } {}
```

Comment:
Why conversion from the base options is needed?

Reply:
because of the implicit conversion [here](https://github.com/KulikovNikita/daal/blob/64a1808dfecbfb2ed213c1118b5575f909b9121a/cpp/oneapi/dal/util/result_option_id.hpp#L70)